### PR TITLE
darwin-framework-tool: Add support for paseonly pairing

### PR DESCRIPTION
**Pull request by ksperling-apple**
This includes code-paseonly and by-index-paseonly commands, as well as interacting with devices over PASE by preferring an existing MTRBaseDevice connected over PASE to creating a new CASE connection when looking up a device by node id.

### CODE REVIEW

**Review by anush-apple — APPROVED** [review]

File: examples/darwin-framework-tool/commands/delay/WaitForCommissioneeCommand.mm [hunk 1]
```diff
@@ -22,10 +22,7 @@
 
 CHIP_ERROR WaitForCommissioneeCommand::RunCommand()
 {
-    MTRDeviceController * commissioner = CurrentCommissioner();
-    VerifyOrReturnError(nil != commissioner, CHIP_ERROR_INCORRECT_STATE);
-
-    auto * base_device = [MTRBaseDevice deviceWithNodeID:@(mNodeId) controller:commissioner];
+    auto * base_device = BaseDeviceWithNodeId(mNodeId);
```
Comment (bzbarsky-apple) [inline examples/darwin-framework-tool/commands/delay/WaitForCommissioneeCommand.mm:25]: Probably does not hurt, but not sure WaitForCommissioneeCommand makes sense over PASE....

File: examples/darwin-framework-tool/commands/pairing/PairingCommandBridge.h [hunk 2]
```diff
@@ -22,15 +22,16 @@
...
     Ble,
     AlreadyDiscoveredByIndex,
 };
 
-enum class PairingNetworkType
+enum class CommissioningType
 {
-    None,
+    None, // establish PASE only
+    NoNetwork,
```
Comment (bzbarsky-apple) [inline examples/darwin-framework-tool/commands/pairing/PairingCommandBridge.h:34]:
```suggestion
    OnNetwork,
```
I would think?

Comment (ksperling-apple) [reply examples/darwin-framework-tool/commands/pairing/PairingCommandBridge.h:34]: No it's mean to mean "Don't configure the network", will rename to something more obvious

Comment (ksperling-apple) [reply examples/darwin-framework-tool/commands/pairing/PairingCommandBridge.h:34]: Will address as a follow-up

**Comment by ksperling-apple** [comment]
Fast-tracking Darwin-only change.

**Pull request merged**